### PR TITLE
Do not let students get verified teacher permissions when they enroll in workshops

### PR DESCRIPTION
**Pull request by dmcavoy**
We have been verifying any account that joins a Workshop or a PLC Course associated to a workshop. However this means we have verified some student accounts. We do not want to be verifying student accounts. As a first step here we are going to stop verifying student accounts that are in a Workshop or PLC Course associated with a workshop. There is follow up work to do here to help facilitators and workshop participants know if they have been verified or not because of their account type. Product is going to follow up on that. However this prevents us from adding more student accounts that are verified in the meantime.

I also found some old code that if you unenrolled from PLC Courses we would remove your verified teacher status. I don't think we want to take that away once someone is verified so removing that.

## Links

[Slack Thread](https://codedotorg.slack.com/archives/CA3KCSGTD/p1643229641056900)

## Testing story

- Unit Tests

### CODE REVIEW

**Review by maureensturgeon — COMMENTED** [review]
What is the risk if users enrolled in workshops don't have authorized teacher permissions? How much confusion will there be?

I also would like to point out that permissions are also assigned here https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/plc/user_course_enrollment.rb#L32
Do we need to check for teacher here as well?

**Comment by dmcavoy** [comment]
> What is the risk if users enrolled in workshops don't have authorized teacher permissions? How much confusion will there be?
> 

Good question Maureen. The risk is that teachers in the workshop might not be able to see teacher only content such as answer keys during the workshop on their account. They can always fill out the form to become verified. Tess is planning to follow up on a way to notifying the user/facilitator if someone is in a student account.

> I also would like to point out that permissions are also assigned here https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/plc/user_course_enrollment.rb#L32 Do we need to check for teacher here as well?

Thanks for point this out. This is for PLC courses associated with workshops which I don't think is the case for any workshops right now but I will also update this path just in case.

File: dashboard/app/models/plc/user_course_enrollment.rb [hunk 1]
```diff
@@ -97,18 +96,8 @@ def summarize
   private
 
   def create_authorized_teacher_user_permission
-    unless user.verified_teacher?
+    if user.teacher? && !user.verified_teacher?
       user.permission = UserPermission::AUTHORIZED_TEACHER
     end
   end
-
-  def delete_authorized_teacher_user_permission
```
Comment (dmcavoy) [inline dashboard/app/models/plc/user_course_enrollment.rb:105]: It does not make sense to take away a teachers verified status once they have received it

**Review by maureensturgeon — APPROVED** [review]
Looks good!

**Review by maureensturgeon — APPROVED** [review]
Looks good!

**Pull request merged**